src/deepEqual.js

```diff
@@ -21,6 +21,37 @@
  */
 function deepEqual(a, b) {
   // write code here
+  let typeOfA = typeof a;
+  let typeOfB = typeof b;
+
+  if (typeOfA === typeOfB && typeOfA !== null && typeOfB !== null) {
+    if (typeOfA === 'object') {
+      if (!isEqualObjectsByKeys(a, b)) {
+        return false;
+      }
+      if (!isEqualObjectsByKeys(b, a)) {
+        return false;
+      }
+      return true;
+    } else {
+      return a === b;
+    }
+  } else {
+    return a === null && b === null;
+  }
+}
+
+function isEqualObjectsByKeys(a, b) {
+  for (let key in a) {
+    if (b !== null) {
+      if (!deepEqual(a[key], b[key])) { // (*) Доступ к ключю a.key видит undefined... Почему так?
```

Comment:
that should not be happening. maybe b[key] is undefined?

Reply:
It was in my browser debugger, I placed own arguments in function... No, it wasn't